### PR TITLE
Remove use of deprecated distutils package

### PR DESCRIPTION
**Pull request by marioalexis84**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6753">pipeline status</a> for feature branch [PR_6753](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6753). Pipeline [517162159](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/517162159) was triggered at [a839658](https://github.com/FreeCAD/FreeCAD/pull/6753/commits/a839658c3d9d9cdb7e5a6071e5f4d9d643665679). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Plot/Plot.py [hunk 1]
```diff
@@ -23,7 +23,6 @@
 
 import PySide
 from PySide import QtCore, QtGui
-from distutils.version import LooseVersion as V
```
Comment (luzpaz) [inline src/Mod/Plot/Plot.py:26]: Is `V` not utilized in the script ?

File: src/Mod/Path/libarea/CMakeLists.txt [hunk 2]
```diff
@@ -166,7 +166,7 @@ SET_PYTHON_PREFIX_SUFFIX(area)
 
 # this figures out where to install the Python modules
 execute_process(
-    COMMAND python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"
+    COMMAND python -c "from sysconfig import get_path; print(get_path(\"purelib\"))"
```
Comment (wwmayer) [inline src/Mod/Path/libarea/CMakeLists.txt:169]: For me `from distutils.sysconfig import get_python_lib; print(get_python_lib())` gives the correct directory /usr/lib/python3/dist-packages while `from sysconfig import get_path; print(get_path("purelib"))` returns the non-existing directory /usr/lib/python3.6/site-packages

File: src/Mod/Plot/Plot.py (hunk 1 above)
Comment (wwmayer) [reply src/Mod/Plot/Plot.py:26]: No.

File: src/Ext/freecad/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c 
-"from distutils.sysconfig import get_python_lib; print(get_python_lib(plat_specific=True))"
```
Comment (wwmayer) [inline src/Ext/freecad/CMakeLists.txt:2]: For me it has the same behaviour as in [src/Mod/Path/libarea/CMakeLists.txt](https://github.com/FreeCAD/FreeCAD/pull/6753/files/a839658c3d9d9cdb7e5a6071e5f4d9d643665679#diff-259df394b9555c95f6c7a3f3e0b045c48931e413e30ae9a5e659b888fedb365a)

**Comment by wwmayer** [comment]
Forum thread: https://forum.freecadweb.org/viewtopic.php?f=10&t=67985

**Comment by donovaly** [comment]
Adding it to the milestone since it is apparently something that should be done before the release.

File: src/Mod/Path/libarea/CMakeLists.txt (hunk 2 above)
Comment (marioalexis84) [reply src/Mod/Path/libarea/CMakeLists.txt:169]: May be related to this: https://wiki.debian.org/Python#Deviations_from_upstream
In Arch Linux I get the path to site-packages for both cases. I will investigate to get the correct path on Debian systems.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6753">pipeline status</a> for feature branch [PR_6753](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6753). Pipeline [518258659](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/518258659) was triggered at [a6b4108](https://github.com/FreeCAD/FreeCAD/pull/6753/commits/a6b4108e57057257140de66ec9b7b4da756c9926). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by marioalexis84** [comment]
See https://forum.freecadweb.org/viewtopic.php?f=10&t=67985&p=589114#p589114
I keep `distutils` in `src/Ext/freecad/CMakeLists.txt` and `src/Mod/Path/libarea/CMakeLists.txt`

**Comment by wwmayer** [comment]
I agree that for the moment we can merge the uncritical part but leave the rest. The `distutils` package will be removed with Py 3.12 but the [current version in development is Py 3.11](https://pythoninsider.blogspot.com/2022/04/the-last-python-311-alpha-3110a7-is.html) and it probably will be another year until Py 3.12 is released. So, no hurry.

**Comment by chennes** [comment]
@donovaly do you still want to merge this before 0.20? It seems less critical now that we are deciding to keep the couple of distutils calls in there for Debian.

**Comment by donovaly** [comment]
> It seems less critical now that we are deciding to keep the couple of distutils calls in there for Debian.

Then fine to merge.

**Comment by yorikvanhavre** [comment]
I would keep this for 0.21 anyway, as it isn't critical for 0.20...

**Comment by donovaly** [comment]
Since I don't know enough, please help me: What are the concerns/risks of not merging it now?

**Comment by wwmayer** [comment]
Parts of the PR could be merged but other parts will cause problems on Debian based systems because the locations where to install packages is different of what Python recommends and what Debian does.

So, the PR may need some rework to correctly handle the specific behaviour on Debian.

> What are the concerns/risks of not merging it now?

I don't see any risk. The distutils package will be removed with Py3.12 but the latest stable versions is Py3.10. When looking at the release dates of Python then there is one version per year and Py3.11 will be available in October 2022 and Py3.12 presumably in October 2023.

So, this gives us plenty of time to sort out all issues and there is no hurry to merge it now.

**Comment by donovaly** [comment]
OK, thanks for explanation.

**Comment by luzpaz** [comment]
I guess we should open another ticket that tracks the issue wwmayer mentions:

> Parts of the PR could be merged but other parts will cause problems on Debian based systems because the locations where to install packages is different of what Python recommends and what Debian does.

Maybe we merge the parts of the PR that work. And separate the other part out in to a separate ticket? What do you think @marioalexis84  ?

**Comment by luzpaz** [comment]
bump

**Comment by donovaly** [comment]
@marioalexis84 , can you please review your changes, merge your commits to one and rebase the PR onto current master?
I think it is now a good time to merge such changes before we enter a new release phase.

**Comment by luzpaz** [comment]
ping @marioalexis84

**Comment by marioalexis84** [comment]
I will look again

**Comment by adrianinsaval** [comment]
has the situation changed regarding this? has debian fixed their python package?

**Comment by wwmayer** [comment]
Not significantly. With Xubuntu 22.04 it returns the path to an existing directory but it's completely empty.

**Comment by berndhahnebach** [comment]
IMHO the FEM commit could be merged separately.

**Comment by luzpaz** [comment]
Is it time to revisit this?  (better to do this before python 3.12 release)

**Comment by luzpaz** [comment]
@chennes :point_up:   
If so then Snap (and possibly flatpak) would need to be tweaked as well.

File: src/Mod/Path/libarea/CMakeLists.txt (hunk 2 above)
Comment (luzpaz) [reply src/Mod/Path/libarea/CMakeLists.txt]: >  I will investigate to get the correct path on Debian systems.  

@marioalexis84 any progress ?

**Comment by adrianinsaval** [comment]
Has Debian figured out their mess?

**Comment by luzpaz** [comment]
@adrianinsaval do you know if there's a Debian ticket we can follow ?

**Comment by chennes** [comment]
We definitely need to do it soon, though, Python 3.12 was released today.

File: src/Mod/Path/libarea/CMakeLists.txt (hunk 2 above)
Comment (adrianinsaval) [reply src/Mod/Path/libarea/CMakeLists.txt]: when you normally build from source I think it is a good idea to put things into site-packages, only debian packagers should be putting stuff into dist-packages (see the above link where they mention why they use dist-packages)

**Comment by wwmayer** [comment]
As this is a Debian-related problem only one could check the package in Debian unstable if its maintainers add a patch to handle the path issue.

**Comment by adrianinsaval** [comment]
debian unstable appears to still be on python 3.11 and freecad 0.20

**Comment by wwmayer** [comment]
> debian unstable appears to still be on python 3.11

Sooner or later they have to make packages for 3.12 and then it becomes interesting to know how they handle the issues.

**Comment by wwmayer** [comment]
For reference: https://peps.python.org/pep-0632/

**Comment by chennes** [comment]
@kkremitzki do you know anything about how Debian is going to handle the Python 3.12 transition?

File: src/Mod/Path/libarea/CMakeLists.txt (hunk 2 above)
Comment (adrianinsaval) [reply src/Mod/Path/libarea/CMakeLists.txt]: on debian sid with python 3.11
```python
>>> from sysconfig import get_path; print(get_path("purelib"))
/usr/local/lib/python3.11/dist-packages
```

Comment (adrianinsaval) [reply src/Mod/Path/libarea/CMakeLists.txt]: ubuntu 22.04 with python 3.10:
```python
>>> from sysconfig import get_path; print(get_path("purelib"))
/usr/local/lib/python3.10/dist-packages
```

Comment (adrianinsaval) [reply src/Mod/Path/libarea/CMakeLists.txt]: and 
```python
>>> from sysconfig import get_path; print(get_path("purelib","deb_system"))
/usr/lib/python3/dist-packages
```

but I think this should only be used by packagers

Comment (adrianinsaval) [reply src/Mod/Path/libarea/CMakeLists.txt]: on 20.04 (which I think we still plan on supporting):
```python
>>> from sysconfig import get_path; print(get_path("purelib","deb_system"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/sysconfig.py", line 525, in get_path
    return get_paths(scheme, vars, expand)[name]
  File "/usr/lib/python3.8/sysconfig.py", line 515, in get_paths
    return _expand_vars(scheme, vars)
  File "/usr/lib/python3.8/sysconfig.py", line 179, in _expand_vars
    for key, value in _INSTALL_SCHEMES[scheme].items():
KeyError: 'deb_system'
>>> from sysconfig import get_path; print(get_path("purelib"))
/usr/lib/python3.8/site-packages
```

**Comment by adrianinsaval** [comment]
we could put distutil calls inside if statements for python versions bellow 3.12 and fully remove it once we drop support for ubuntu 20.04 in 2025

**Pull request closed without merging**